gnd/src/main/java/com/google/android/gnd/persistence/local/room/LayerData.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.local.room;
+
+import androidx.room.Embedded;
+import androidx.room.Relation;
+import java.util.List;
+
+/**
+ * Represents relationship between LayerEntity and FormEntity.
+ *
+ * <p>Querying any of the below data class automatically loads the field annotated as @Relation.
+ */
+public class LayerData {
```

Comment:
Same here. Do you mean these are being lazily loading from the db when you iterate or access the list?

Reply:
It doesn't load the data lazily. The point of having a separate class is to load LayerEntity and all FormData (which is basically FormEntity with similar relations)

Reply:
Gotcha. Thanks for bearing with me! See previous comment.